catalog/core/catalog-core-backupplugin/src/main/java/ddf/catalog/backup/CatalogBackupPlugin.java

```diff
@@ -167,42 +108,44 @@ public UpdateResponse process(UpdateResponse input) throws PluginExecutionExcept
      * Removes deleted metacards from the file system backup.
      *
      * @param input the {@link DeleteResponse} to process
-     * @return
-     * @throws PluginExecutionException
+     * @return {@link DeleteResponse}
      */
     @Override
-    public DeleteResponse process(DeleteResponse input) throws PluginExecutionException {
-        if (enableBackupPlugin && Requests.isLocal(input.getRequest())) {
-            LOGGER.debug("Deleting metacards contained in DeleteResponse from backup.");
+    public DeleteResponse process(DeleteResponse input) {
 
-            if (rootBackupDir == null) {
-                throw new PluginExecutionException("No root backup directory configured.");
-            }
-
-            List<String> errors = new ArrayList<String>();
+        execute(() -> delete(input.getDeletedMetacards()));
+        return input;
+    }
 
-            List<Metacard> metacards = input.getDeletedMetacards();
+    /**
+     * Sets the number of subdirectory levels to create. Two characters from
+     * each metacard ID will be used to name each subdirectory level.
+     *
+     * @param levels number of subdirectory levels to create
+     */
+    public void setSubDirLevels(int levels) {
+        this.subDirLevels = levels;
+        LOGGER.debug("Set subdirectory levels to: {}", this.subDirLevels);
+    }
 
-            for (Metacard metacard : metacards) {
-                try {
-                    deleteMetacard(metacard);
-                } catch (IOException e) {
-                    errors.add(metacard.getId());
-                }
+    @SuppressWarnings("unused")
+    public void shutdown() {
+
+        getExecutor().shutdown();
+        if (!executor.isShutdown()) {
+            try {
+                executor.awaitTermination(getTerminationTimeoutSeconds(), TimeUnit.SECONDS);
+                // If the remaining jobs in the queue are not completed TERMINATION_TIMEOUT elapses
+                // The JVM will throw an Interrupted Exception.
+            } catch (InterruptedException e) {
+                throw new IllegalStateException(e);
             }
-
-            if (errors.size() > 0) {
-                throw new PluginExecutionException(getExceptionMessage(DeleteResponse.class.getSimpleName(),
-                        null,
-                        errors,
-                        OPERATION.DELETE));
+            final List<Runnable> failures = executor.shutdownNow();
+            final int failedCount = failures.size();
+            if (failedCount > 0) {
+                LOGGER.debug("Failed to execute {} tasks", failedCount);
```

Comment:
If this happens, it means that some metacards weren't backed up as expected so we may want to make this at least a warning.